### PR TITLE
Hardcoded LoginView

### LINKED ISSUE #416: Hardcoded LoginView
https://github.com/pydanny/django-admin2/blob/develop/djadmin2/viewmixins.py#L104-L113

### PR DESCRIPTION
**Pull request by arthur-wsw**
Add a way to customize LoginView
Add documentation on how tow customize login view

Resolve #416

### CODE REVIEW

**Comment by galuszkak** [comment]
@arthur-wsw it will be good to have some test using custom login view.

**Comment by arthur-wsw** [comment]
I wasn't sure on how it can be tested, does anyone has an idea on how the test could be written ?

**Comment by arthur-wsw** [comment]
I finally was able to add a test by overriding the ROOT_URLCONF settings

@galuszkak i just added a test that check html in response. Is this test enough?

File: djadmin2/tests/test_views.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
-from django.test import TestCase
+from django.core.urlresolvers import reverse
+from django.test import TestCase, override_settings
+from django.utils.encoding import force_text
+from os import path
 
 from .. import views
 
+fixture_dir = path.join(path.abspath(path.dirname(__file__)), 'fixtures')
+fixture_file = path.join(fixture_dir, 'pubtest.txt')
```
Comment (andrewsmedina) [inline djadmin2/tests/test_views.py]: where these fixtures are used?

Comment (arthur-wsw) [inline djadmin2/tests/test_views.py]: @andrewsmedina Probably nowhere...

I just forgot to delete them after a copy/paste, it should now be better

File: docs/ref/views.rst [hunk 2]
```diff
@@ -44,4 +44,39 @@ In your Django project's root URLconf module (``urls.py``) modify the code to in
...
+Customizing the Login view
+==========================
+
+The login view could also be customized.
+
+In your Django project's root URLconf module (``urls.py``) modify the code to include the commented code before the ``djadmin2.default.autodiscover()``:
+
+.. code-block:: python
+
+    from django.conf.urls import patterns, include, url
```
Comment (auvipy) [inline docs/ref/views.rst]: patterns are deprecated

File: docs/ref/views.rst [hunk 3]
```diff
@@ -44,4 +44,39 @@ In your Django project's root URLconf module (``urls.py``) modify the code to in
...
+        # specify the template
+        default_template_name = "custom_login_template.html"
+
+    # override the default index_view
+    djadmin2_site.login_view = CustomLoginView
+    ######### end django-admin2 customization code
+
+    djadmin2_site.autodiscover()
+
+    urlpatterns = patterns('',
```
Comment (auvipy) [inline docs/ref/views.rst]: better switch to 1.8+ [
url()
]

Comment (arthur-wsw) [inline docs/ref/views.rst]: @auvipy Thanks for notified that, i have fix it on #439

**Comment by auvipy** [comment]
@arthur-wsw could u pull the changes and re-base?

**Pull request merged**